arch/arm/arm64/Makefile.uk

```diff
@@ -17,57 +17,188 @@ ASINCLUDES  += -I$(CONFIG_UK_BASE)/arch/arm/arm64/include
+ARCHFLAGS-$(call gcc_version_ge,9,1)     += -mtune=cortex-a76.cortex-a55
+ISR_ARCHFLAGS-$(call gcc_version_ge,9,1) += -mcpu=cortex-a76.cortex-a55
+ISR_ARCHFLAGS-$(call gcc_version_ge,9,1) += -mtune=cortex-a76.cortex-a55
+endif
+
+# GCC support -mcpu=neoverse-n1 for arm64 from 9.1
+ifeq ($(CONFIG_MCPU_ARM64_NEOVERSE_N1),y)
+$(call error_if_gcc_version_lt,9,1)
+ARCHFLAGS-$(call gcc_version_ge,9,1)     += -mcpu=neoverse-n1 -mtune=neoverse-n1
+ISR_ARCHFLAGS-$(call gcc_version_ge,9,1) += -mcpu=neoverse-n1 -mtune=neoverse-n1
+endif
+
+# GCC support -mcpu=neoverse-e1 for arm64 from 9.1
+ifeq ($(CONFIG_MCPU_ARM64_NEOVERSE_E1),y)
+$(call error_if_gcc_version_lt,9,1)
+ARCHFLAGS-$(call gcc_version_ge,9,1)     += -mcpu=neoverse-e1 -mtune=neoverse-e1
+ISR_ARCHFLAGS-$(call gcc_version_ge,9,1) += -mcpu=neoverse-e1 -mtune=neoverse-e1
+endif
+
+# GCC support -mcpu=cortex-a34 for arm64 from 10.1
+ifeq ($(CONFIG_MCPU_ARM64_CORTEX_A34),y)
+$(call error_if_gcc_version_lt,10,1)
+ARCHFLAGS-$(call gcc_version_ge,10,1)     += -mcpu=cortex-a34 -mtune=cortex-a34
+ISR_ARCHFLAGS-$(call gcc_version_ge,10,1) += -mcpu=cortex-a34 -mtune=cortex-a34
+endif
+
+# GCC support -mcpu=cortex-a65 for arm64 from 10.1
+ifeq ($(CONFIG_MCPU_ARM64_CORTEX_A65),y)
+$(call error_if_gcc_version_lt,10,1)
+ARCHFLAGS-$(call gcc_version_ge,10,1)     += -mcpu=cortex-a65 -mtune=cortex-a65
+ISR_ARCHFLAGS-$(call gcc_version_ge,10,1) += -mcpu=cortex-a65 -mtune=cortex-a65
+endif
+
+# GCC support -mcpu=cortex-a65ae for arm64 from 10.1
+ifeq ($(CONFIG_MCPU_ARM64_CORTEX_A65AE),y)
+$(call error_if_gcc_version_lt,10,1)
+ARCHFLAGS-$(call gcc_version_ge,10,1)     += -mcpu=cortex-a65ae -mtune=cortex-a65ae
+ISR_ARCHFLAGS-$(call gcc_version_ge,10,1) += -mcpu=cortex-a65ae -mtune=cortex-a65ae
+endif
+
+# GCC support -mcpu=cortex-a76ae for arm64 from 10.1
+ifeq ($(CONFIG_MCPU_ARM64_CORTEX_A76AE),y)
+$(call error_if_gcc_version_lt,10,1)
+ARCHFLAGS-$(call gcc_version_ge,10,1)     += -mcpu=cortex-a76ae -mtune=cortex-a76ae
+ISR_ARCHFLAGS-$(call gcc_version_ge,10,1) += -mcpu=cortex-a76ae -mtune=cortex-a76ae
+endif
+
+# GCC support -mcpu=cortex-a77 for arm64 from 10.1
+ifeq ($(CONFIG_MCPU_ARM64_CORTEX_A77),y)
+$(call error_if_gcc_version_lt,10,1)
+ARCHFLAGS-$(call gcc_version_ge,10,1)     += -mcpu=cortex-a77 -mtune=cortex-a77
+ISR_ARCHFLAGS-$(call gcc_version_ge,10,1) += -mcpu=cortex-a77 -mtune=cortex-a77
+endif
+
+# GCC support -mcpu=neoverse-n2 for arm64 from 10.1
+ifeq ($(CONFIG_MCPU_ARM64_NEOVERSE_N2),y)
+$(call error_if_gcc_version_lt,10,1)
+ARCHFLAGS-$(call gcc_version_ge,10,1)     += -mcpu=neoverse-n2 -mtune=neoverse-n2
+ISR_ARCHFLAGS-$(call gcc_version_ge,10,1) += -mcpu=neoverse-n2 -mtune=neoverse-n2
+endif
+
+# GCC support -mcpu=neoverse-v1 for arm64 from 10.1
+ifeq ($(CONFIG_MCPU_ARM64_NEOVERSE_V1),y)
+$(call error_if_gcc_version_lt,10,1)
+ARCHFLAGS-$(call gcc_version_ge,10,1)     += -mcpu=neoverse-v1 -mtune=neoverse-v1
+ISR_ARCHFLAGS-$(call gcc_version_ge,10,1) += -mcpu=neoverse-v1 -mtune=neoverse-v1
 endif
```

Comment:
What about `gcc-11`? It recently came out and it would be good to also have it here.

Reply:
It's only because GCC 11 has not landed yet in https://developer.arm.com/tools-and-software/open-source-software/developer-tools/gnu-toolchain/gnu-a/downloads :)

Reply:
Whoops, I thought gcc versions are released at the same time for all architectures 😅

Reply:
They are :) It just happens that Arm has not released an official build yet. Once they do I will make a PR for the newly supported features.